### ISSUE TITLE
Not properly handling wasted bits

### ISSUE THREAD

**Issue by ktmf01** [astoeckel/libfoxenflac#4]
I just found your FLAC decoding library, and tested it with [this FLAC decoder testbench](https://github.com/ietf-wg-cellar/flac-test-files/tree/main/subset). All files decode fine, except number 14.

Still, this is quite a feat, almost all decoders I tried fail one or a few files. I hope this helps you to even further improve libfoxenflac.

**Comment by astoeckel** [comment]
Thank you for testing and reporting! Will have a look in the near future :-)

**Comment by astoeckel** [comment]
Yay, I found and fixed the bug!

I only applied the wasted bit transformation after all subframes of a frame had been processed. This didn't account for the wasted bit count changing within a subframe.

I also added the complete testbench to the integration tests, which uncovered another bug. I mistyped "1024" as "1204" in two places, which was causing a file with variable block sizes to fail (blocks following blocks of size 1024 were skipped‒the decoder noticed that something was wrong and resynchronised).

Thanks again for pointing out this awesome resource! For some reason I didn't find this back in 2018...
